appinventor/components/src/com/google/appinventor/components/runtime/Chart.java

```diff
@@ -0,0 +1,276 @@
+        Height(ComponentConstants.VIDEOPLAYER_PREFERRED_HEIGHT);
+        BackgroundColor(Component.COLOR_DEFAULT);
+        Description("");
+    }
+
+    @Override
+    public View getView() {
+        return view;
+    }
+
+    @Override
+    public Activity $context() {
+        return container.$context();
+    }
+
+    @Override
+    public Form $form() {
+        return container.$form();
+    }
+
+    @Override
+    public void $add(AndroidViewComponent component) {
+        throw new UnsupportedOperationException("ChartBase.$add() called");
+    }
+
+    @Override
+    public void setChildWidth(AndroidViewComponent component, int width) {
+        throw new UnsupportedOperationException("ChartBase.setChildWidth called");
+    }
+
+    @Override
+    public void setChildHeight(AndroidViewComponent component, int height) {
+        throw new UnsupportedOperationException("ChartBase.setChildHeight called");
+    }
+
+    /**
+     * Returns the type of the Chart.
+     *
+     * @return  one of {@link Component#CHART_TYPE_LINE},
+     *          {@link Component#CHART_TYPE_SCATTER},
+     *          {@link Component#CHART_TYPE_AREA},
+     *          {@link Component#CHART_TYPE_BAR} or
+     *          {@link Component#CHART_TYPE_PIE}
+     */
+    @SimpleProperty(
+            category = PropertyCategory.BEHAVIOR,
+            userVisible = false)
+    public int Type() {
+        return type;
+    }
+
+    /**
+     * Specifies the type of the Chart. This does not check that the argument is a legal value.
+     *
+     * @param type one of {@link Component#CHART_TYPE_LINE},
+     *  {@link Component#CHART_TYPE_SCATTER},
+     *  {@link Component#CHART_TYPE_AREA},
+     *  {@link Component#CHART_TYPE_BAR} or
+     *  {@link Component#CHART_TYPE_PIE}
+     *
+     * @throws IllegalArgumentException if shape is not a legal value.
+     */
+    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_CHART_TYPE,
+            defaultValue = Component.CHART_TYPE_LINE + "")
+    @SimpleProperty(description = "Specifies the chart's type (line, scatter," +
+            "area, bar, pie).",
+            userVisible = false)
+    public void Type(int type) {
+        this.type = type;
+
+        // Keep track whether a ChartView already exists,
+        // in which case it will have to be reinitialized.
+        boolean chartViewExists = (chartView != null);
+
+        // ChartView currently exists in root layout. Remove it.
+        if (chartViewExists) {
+            view.removeView(chartView.getView());
+        }
+
+        // Create a new Chart view based on the supplied type
+        chartView = createChartViewFromType(type);
+
+        // Add the new Chart view as the first child of the root RelativeLayout
+        view.addView(chartView.getView(), 0);
+
+        // If a ChartView already existed, then the Chart
+        // has to be reinitialized.
+        if (chartViewExists) {
```

Comment:
The Type property chosen in Designer (if it is not the default value) gets registered only after all the Data components are already attached to the Chart, hence, they have to be re-added.

Any suggestions of a different, simpler approach, or is this fine?